app/models/tutorial.rb

```diff
@@ -14,16 +14,20 @@ def playlist_video_json
   end
 
   def playlist_video_params
-    playlist_video_params = []
-    playlist_video_json[:items].each do |video|
-      video_params = {}
-      video_params[:title] = video[:snippet][:title]
-      video_params[:description] = video[:snippet][:description]
-      video_params[:video_id] = video[:contentDetails][:videoId]
-      video_params[:thumbnail] = video[:snippet][:thumbnails][:default][:url]
-      video_params[:position] = video[:snippet][:position]
-      playlist_video_params << video_params
+    playlist_video_json[:items].map do |video|
+      Hash[playlist_video_params_keys.zip(playlist_video_params_values(video))]
     end
```

Comment:
given keys and values, zip together on each iteration to create an array of video_param hashes

Reply:
nice! gotta love that zip!